packages/block-library/src/buttons/edit.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import { IconButton } from '@wordpress/components';
+import { InnerBlocks } from '@wordpress/block-editor';
+import { createBlock } from '@wordpress/blocks';
+import { useCallback } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { name as buttonBlockName } from '../button/';
+import { ButtonEditSettings } from '../button/edit-settings';
+
+const ALLOWED_BLOCKS = [ buttonBlockName ];
+const BUTTON_BLOCK_SETTINGS = { urlInPopover: true };
+const BUTTONS_TEMPLATE = [ [ 'core/button' ] ];
+const UI_PARTS = {
+	hasSelectedUI: false,
+	hasFocusedUI: false,
+	hasHoveredUI: false,
+	hasBreadcrumbs: false,
+	hasMovers: false,
+	hasSpacing: false,
+	hasSideInserter: false,
+};
+
+function useInsertButton( clientId ) {
+	const { insertBlock } = useDispatch( 'core/block-editor' );
+	const insertButton = useCallback(
+		() => {
+			const buttonBlock = createBlock( buttonBlockName );
+			insertBlock( buttonBlock, undefined, clientId );
+		},
+		[ insertBlock, clientId ]
+	);
+	return useCallback(
+		() => {
+			return (
+				<IconButton
+					icon="insert"
+					label={ __( 'Add button' ) }
+					labelPosition="bottom"
+					onClick={ insertButton }
+				/>
+			);
+		},
+		[ insertButton ]
+	);
+}
+
+function ButtonsEdit( { clientId, className } ) {
+	const renderAppender = useInsertButton( clientId );
```

Comment:
@draganescu worked on a PR where if there's only one allowed child block, the appender is automatically something very similar to the one used here. Can't we just rely on this behavior?

Also, I feel this is also very similar to the Navigation block, how is this solved there?

Reply:
Yes, this code is not necessary as this behavior is now the default one, if only one block is allowed it is inserted right away. This code was added before this functionality was merged and it was now removed.